modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java

```diff
@@ -135,6 +135,15 @@ public void start() {
 
                 connectionMgr.start();
 
+                topologyService.addEventHandler(new TopologyEventHandler() {
+                    private final ConnectionManager capturedConnectionManager = connectionMgr;
```

Comment:
Not a big deal, but maybe we should extract all volatile fields into local variables and only assign them at the end of the method? This way we can avoid declaring this field and possibly avoid volatile reads.